sklearn/preprocessing/imputation.py

```diff
@@ -357,6 +379,17 @@ def transform(self, X):
 
             X.data[mask] = astype(valid_statistics[indexes], X.dtype,
                                   copy=False)
+
+            if self.add_missing_indicator:
+                mask_matrix = X.__class__((mask, X.indices.copy(), X.indptr.copy()), shape=X.shape, dtype=X.dtype)
+                mask_matrix.eliminate_zeros()  # removes explicit False entries to make the matrix sparser
+                features_with_missing_values = np.where(np.logical_not(sparse_min_max(mask_matrix, 0)[1]))
```

Comment:
Sorry I think I got confused. Thanks I understood the idea. Also ``np.where(...)[0]`` since it is a 1-D array and the tuple is of size 1. Hope I am understanding it right.